apps/hash-integration-worker/src/mappings.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  Attachment,
+  Comment,
+  CustomView,
+  Cycle,
+  Document,
+  Issue,
+  IssueLabel,
+  Organization,
+  Project,
+  ProjectMilestone,
+  Team,
+  User,
+} from "@linear/sdk";
+import { linearTypes } from "@local/hash-isomorphic-utils/ontology-types";
+import { extractBaseUrl } from "@local/hash-subgraph/type-system-patch";
+
+export const userToEntity = (user: User): object => {
+  return {
+    [extractBaseUrl(linearTypes.propertyType.active.propertyTypeId)]:
+      user.active,
+    [extractBaseUrl(linearTypes.propertyType.admin.propertyTypeId)]: user.admin,
+    [extractBaseUrl(linearTypes.propertyType.archivedAt.propertyTypeId)]:
+      user.archivedAt,
+    [extractBaseUrl(linearTypes.propertyType.avatarUrl.propertyTypeId)]:
+      user.avatarUrl,
+    [extractBaseUrl(linearTypes.propertyType.createdAt.propertyTypeId)]:
+      user.createdAt,
+    [extractBaseUrl(linearTypes.propertyType.createdIssueCount.propertyTypeId)]:
+      user.createdIssueCount,
+    [extractBaseUrl(linearTypes.propertyType.description.propertyTypeId)]:
+      user.description,
+    [extractBaseUrl(linearTypes.propertyType.disableReason.propertyTypeId)]:
+      user.disableReason,
+    [extractBaseUrl(linearTypes.propertyType.displayName.propertyTypeId)]:
+      user.displayName,
+    [extractBaseUrl(linearTypes.propertyType.email.propertyTypeId)]: user.email,
+    [extractBaseUrl(linearTypes.propertyType.guest.propertyTypeId)]: user.guest,
+    [extractBaseUrl(linearTypes.propertyType.id.propertyTypeId)]: user.id,
+    [extractBaseUrl(linearTypes.propertyType.isMe.propertyTypeId)]: user.isMe,
+    [extractBaseUrl(linearTypes.propertyType.lastSeen.propertyTypeId)]:
+      user.lastSeen,
+    [extractBaseUrl(linearTypes.propertyType.statusEmoji.propertyTypeId)]:
+      user.statusEmoji,
+    [extractBaseUrl(linearTypes.propertyType.statusLabel.propertyTypeId)]:
+      user.statusLabel,
+    [extractBaseUrl(linearTypes.propertyType.statusUntilAt.propertyTypeId)]:
+      user.statusUntilAt,
+    [extractBaseUrl(linearTypes.propertyType.timezone.propertyTypeId)]:
+      user.timezone,
+    [extractBaseUrl(linearTypes.propertyType.updatedAt.propertyTypeId)]:
+      user.updatedAt,
+    [extractBaseUrl(linearTypes.propertyType.url.propertyTypeId)]: user.url,
+  };
+};
```

Comment:
We could use `Entity["properties"]` as a type rather than `object` which is more correct, and encourages us to be explicit about changing types from the SDK such as `Date` to their JSON representation – in this case explicitly calling `.toISOString()` happens to be the same behaviour as the implicit conversion, but it might catch other non-JSON serializable types that the SDK adds

```suggestion
import { Entity } from "@local/hash-subgraph";

...

export const userToEntity = (user: User): Partial<Entity["properties"]> => {
  return {
    [extractBaseUrl(linearTypes.propertyType.active.propertyTypeId)]:
      user.active,
    [extractBaseUrl(linearTypes.propertyType.admin.propertyTypeId)]: user.admin,
    [extractBaseUrl(linearTypes.propertyType.archivedAt.propertyTypeId)]:
      user.archivedAt?.toISOString(),
    [extractBaseUrl(linearTypes.propertyType.avatarUrl.propertyTypeId)]:
      user.avatarUrl,
    [extractBaseUrl(linearTypes.propertyType.createdAt.propertyTypeId)]:
      user.createdAt.toISOString(),
    [extractBaseUrl(linearTypes.propertyType.createdIssueCount.propertyTypeId)]:
      user.createdIssueCount,
    [extractBaseUrl(linearTypes.propertyType.description.propertyTypeId)]:
      user.description,
    [extractBaseUrl(linearTypes.propertyType.disableReason.propertyTypeId)]:
      user.disableReason,
    [extractBaseUrl(linearTypes.propertyType.displayName.propertyTypeId)]:
      user.displayName,
    [extractBaseUrl(linearTypes.propertyType.email.propertyTypeId)]: user.email,
    [extractBaseUrl(linearTypes.propertyType.guest.propertyTypeId)]: user.guest,
    [extractBaseUrl(linearTypes.propertyType.id.propertyTypeId)]: user.id,
    [extractBaseUrl(linearTypes.propertyType.isMe.propertyTypeId)]: user.isMe,
    [extractBaseUrl(linearTypes.propertyType.lastSeen.propertyTypeId)]:
      user.lastSeen?.toISOString(),
    [extractBaseUrl(linearTypes.propertyType.statusEmoji.propertyTypeId)]:
      user.statusEmoji,
    [extractBaseUrl(linearTypes.propertyType.statusLabel.propertyTypeId)]:
      user.statusLabel,
    [extractBaseUrl(linearTypes.propertyType.statusUntilAt.propertyTypeId)]:
      user.statusUntilAt?.toISOString(),
    [extractBaseUrl(linearTypes.propertyType.timezone.propertyTypeId)]:
      user.timezone,
    [extractBaseUrl(linearTypes.propertyType.updatedAt.propertyTypeId)]:
      user.updatedAt.toISOString(),
    [extractBaseUrl(linearTypes.propertyType.url.propertyTypeId)]: user.url,
  };
};
```